### PR TITLE
Implement the fn:lang function

### PR DESCRIPTION
**Pull request by egh**
- Implemented the [`lang`](https://www.w3.org/TR/xpath-functions-31/#func-lang) function.
- Added some basic tests for this
- Not sure how to properly run the QT3 test suite to test this - I ran the test suite and everything passed, but with pending tests.

### CODE REVIEW

**Comment by DrRataplan** [comment]
Hey Erik,

Thanks so much! I'll fix some small things when I get home again, but to answer your questions:

- The QT3 tests can be done using the command `npm run alltests -- --regenerate`. This will empty out the known-failing list and refill it with new errors. It takes some time though, so make sure you can get a coffee while waiting :wink:
- But before that, the fn-lang-related tests should be marked as possibly running. You can do that here: https://github.com/FontoXML/fontoxpath/blob/master/test/assets/runnableTestSets.csv#L114. Just set it to `true` before running the tests.
- I see some small linting / formatting errors. Pesky tabs versus spaces. A `npm run lint-fix` should resolve that.

I'll give it a review when I get home, but looks great! And thanks so much for the tests!

Kindest regards,

Martin

**Comment by egh** [comment]
> Hey Erik,
> 
> Thanks so much! I'll fix some small things when I get home again, but to answer your questions:
> 
>     * The QT3 tests can be done using the command `npm run alltests -- --regenerate`. This will empty out the known-failing list and refill it with new errors. It takes some time though, so make sure you can get a coffee while waiting 😉
> 
>     * But before that, the fn-lang-related tests should be marked as possibly running. You can do that here: https://github.com/FontoXML/fontoxpath/blob/master/test/assets/runnableTestSets.csv#L114. Just set it to `true` before running the tests.
> 
>     * I see some small linting / formatting errors. Pesky tabs versus spaces. A `npm run lint-fix` should resolve that.
> 
> 
> I'll give it a review when I get home, but looks great! And thanks so much for the tests!


I fixed the linting problems, enabled & ran the QT3 tests, and fixed the bugs it revealed. There are 2 outstanding `lang` tests that are failing, but I frankly don't understand the xpath so I'm sure what is going wrong.

**Comment by DrRataplan** [comment]
Hey Erik,

Thanks! The test that is still failing is of the shape `let $l := /langs/para[4]!fn:lang#1 return /langs/para[1]!$l('de')`. I think it's trying to bind the context item of paragraph 4 to the fn:lang function. I'll have to look further into that, but it looks like the function binding is misbehaving here, rather than the lang function.

This does not block merging.

I'll wait until the pipeline is done and I'll merge it!

Thanks again!

Martin

**Review by DrRataplan — APPROVED** [review]

File: src/expressions/functions/builtInFunctions_node.ts
Comment (bwrrp) [inline src/expressions/functions/builtInFunctions_node.ts:194]: Should this be a namespace-aware getAttributeNS?

File: src/expressions/functions/builtInFunctions_node.ts
Comment (bwrrp) [inline src/expressions/functions/builtInFunctions_node.ts:177]: Are we certain no valid tags are prefixes of each other?
Also, why 5? Maybe checking for the presence of a `-` is a better exit cause for the recursion.

File: src/expressions/functions/builtInFunctions_node.ts
Comment (egh) [reply src/expressions/functions/builtInFunctions_node.ts:194]: Weirdly, it's required to use the `xml` prefix, unlike other namespaces. https://www.w3.org/XML/1998/namespace

File: src/expressions/functions/builtInFunctions_node.ts
Comment (egh) [reply src/expressions/functions/builtInFunctions_node.ts:177]: The `startswith` is just a heuristic, the real testing is done after splitting off the last part.

`< 5` because it will match `en-us` and won't match `en`. Maybe it should be `truth.contains("-")`

File: src/expressions/functions/builtInFunctions_node.ts
Comment (bwrrp) [reply src/expressions/functions/builtInFunctions_node.ts:194]: Ah true, that would make this equivalent 👍

**Review by bwrrp — APPROVED** [review]

**Pull request merged**
